test/tests/time_steppers/iteration_adaptive/tests

```diff
@@ -62,6 +62,16 @@
     when specifying the values of t and dt in the TimeStepper after a restart."
   []
 
+  [adapt_tstep_linear_system]
+    type = 'CSVDiff'
+    input = 'adapt_linear_systems.i'
+    csvdiff = 'adapt_linear_systems_out.csv'
+    requirement = "The system shall support the ability to adapt the time step size based on the number of iterations from linear systems."
+    # Number of linear iterations varies a lot in parallel
+    max_parallel = 1
```

Comment:
This is a warning bell for me. Tests that are essentially iteration counters are a recipe for failure with changing architectures (e.g. GPU), slight tweaks in PETSc algorithms, etc. I would be ok with using something like `-pc_type lu -pc_factor_mat_solver_type mumps` which we can reasonably trust to always give the same number of iterations

Reply:
we would need it to take more than 1 iteration though.

I might be able to cook something up with a Convergence object

Reply:
ok removed. 
@joshuahansel  I ended up modifying the convergence code to be able to use it here. 
What I did is not the only option we could:
- spring up a linear convergence parameter, that we then use both for linear systems and KSP-nested instead nonlinear systems
- instead of calling nonlinearxyz -> solverxyz (what I did, following what Peter did for adding linear systems), we could call it: outerSolver, SolverSystem, ... . Is SolverConvergence the best name? Maybe not. SolverSystemConvergence is a tad more accurate

Reply:
Discussed offline; current plan would be to keep `nonlinear_convergence` and add `linear_convergence`, as this would be more consistent with Executioner params: `nonlinear_convergence` is ignored if you're not doing any nonlinear solve. I think it's easier for users to understand than `outer/inner_convergence`.

Reply:
outer/inner is not the best for sure. 
The only concern with opening up "linear" is that this means more than my use case and we ll just put an error outside of my use case for now